src/baldr/edge_elevation.cc

```diff
@@ -0,0 +1,53 @@
+#include <cmath>
+#include <valhalla/midgard/logging.h>
+#include "baldr/edge_elevation.h"
+
+namespace valhalla {
+namespace baldr {
+
+// Constructor with arguments.
+EdgeElevation:: EdgeElevation(const float mean_elev, const float max_up_slope,
+                              const float max_down_slope) {
+  set_mean_elevation(mean_elev);
+  set_max_up_slope(max_up_slope);
+  set_max_down_slope(max_down_slope);
+}
+
+// Set the mean elevation.
+void EdgeElevation::set_mean_elevation(const float mean_elev) {
```

Comment:
thinking here what will happen if skadi returns `NO_DATA_VALUE` for sampled heights  (or if tiles are built without elevation data) - this gets then clamped to `kMinElevation`. wonder if it's better to introduce an explicit separate sentinel value (`kNoElevationData`) - this will be returned from `EdgeElevation::elevation()` when there's no data. That is:

`mean_elevation_ = 0` means `kNoElevationData`
`mean_elevation_ = 1` means `kMinElevation`
`kElevationNoData = -32768`

In other words something along the lines:

```
if (mean_elev > kMaxElevation || mean_elev < kMinElevation)
  mean_elevation_ = 0;
else
 mean_elevation_ = (mean_elev - (kMinElevation - 1)) / kElevationBinSize;
```

and for `elevation()` something like:
`return elevation_ > 0 ? ((mean_elevation_ * kElevationBinSize) + (kMinElevation - 1)) : kElevationNoData;`

Thoughts?